config/src/create_genesis.ts

```diff
@@ -36,6 +38,15 @@ export async function generateGenesisFile(
   baseGenesis.preDeploy = undefined;
   baseGenesis.alloc = Object.fromEntries(alloc);
 
+  try {
+    const provider = new ethers.providers.WebSocketProvider(l1Network)
+    const block = await provider.getBlock("safe")
+    baseGenesis.timestamp = block.timestamp
+    provider._websocket.terminate();
```

Comment:
ethers v6 provides a better handle to close the websocket, but we need to do this in v5 so the script terminates https://github.com/ethers-io/ethers.js/issues/1053